homeassistant/components/august/config_flow.py

```diff
@@ -87,30 +78,45 @@ async def async_step_user(self, user_input=None):
             self._august_gateway = AugustGateway(self.hass)
         errors = {}
         if user_input is not None:
-            await self._august_gateway.async_setup(user_input)
+            combined_inputs = {**self.user_auth_details, **user_input}
+            await self._august_gateway.async_setup(combined_inputs)
+            if self._needs_reset:
+                self._needs_reset = False
+                await self._august_gateway.async_reset_authentication()
 
             try:
                 info = await async_validate_input(
-                    self.hass,
-                    user_input,
+                    combined_inputs,
                     self._august_gateway,
                 )
-                await self.async_set_unique_id(user_input[CONF_USERNAME])
-                return self.async_create_entry(title=info["title"], data=info["data"])
             except CannotConnect:
                 errors["base"] = "cannot_connect"
             except InvalidAuth:
                 errors["base"] = "invalid_auth"
             except RequireValidation:
-                self.user_auth_details = user_input
+                self.user_auth_details.update(user_input)
 
                 return await self.async_step_validation()
             except Exception:  # pylint: disable=broad-except
                 _LOGGER.exception("Unexpected exception")
                 errors["base"] = "unknown"
 
+            if not errors:
+                self.user_auth_details.update(user_input)
+
+                existing_entry = await self.async_set_unique_id(
+                    combined_inputs[CONF_USERNAME]
+                )
+                if existing_entry:
```

Comment:
Is this only the case if we're in a reauth flow?

Reply:
Yes